es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
right - but this table *will, guaranteed* be extended in the future to add more primitive types. SIMD is stage 3, soon to be stage 4, and defines like 20 of them. As-is, that means `es6-shim` will have to add special cases for all SIMD values, and any intervening versions that contain this change will be broken in future implementations that expose the new primitives.